test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -712,7 +724,9 @@ protected <T extends AggregationBuilder, V extends InternalAggregation> void tes
     ) throws IOException {
         boolean timeSeries = aggTestConfig.builder().isInSortOrderExecutionRequired();
         try (Directory directory = newDirectory()) {
-            IndexWriterConfig config = LuceneTestCase.newIndexWriterConfig(random(), new MockAnalyzer(random()));
+            // Use LogDocMergePolicy to avoid randomization issues with the doc retrieval order.
```

Comment:
I don't think we want to do this for all the tests. This is only an issue for nested aggregations and should only be done in such a case.